roles/epel/tasks/main.yml

```diff
@@ -18,11 +18,15 @@
     name='https://dl.fedoraproject.org/pub/epel/epel-release-latest-{{ ansible_distribution_major_version  }}.noarch.rpm'
     state=present
 
+- name: Check if EPEL is not permanently enabled on RHEL
+  fail: msg="Do not permanently enable EPEL on RHEL!"
+  when: enabled and ansible_distribution == 'RedHat'
+
```

Comment:
Could you move this block to the beginning of the role, so that it would fail immediately not leaving the epel repo enabled after the fail?